pkg/logs/logs.go

```diff
@@ -116,46 +114,41 @@ func (o *LogsClient) getPodsForSelector(
 	namespace string,
 	podChan chan corev1.Pod,
 ) error {
-	resources, err := o.kubernetesClient.GetAllResourcesFromSelector(selector, namespace)
+	// set of unique Pods with Pod name as key; these are the Pods whose logs we want to get from the cluster
+	pods := map[string]struct{}{}
+
+	podList, err := o.kubernetesClient.GetPodsMatchingSelector(selector)
 	if err != nil {
 		return err
 	}
-	// set of unique Pods with Pod name as key; these are the Pods whose logs we want to get from the cluster
-	pods := map[string]struct{}{}
+	for _, pod := range podList.Items {
+		podChan <- pod
+		pods[pod.GetName()] = struct{}{}
+	}
 
-	// if there's a Pod in the resources, we add it to the set of Pods whose logs we are interested in
-	for _, r := range resources {
-		if r.GetKind() == "Pod" {
-			var pod corev1.Pod
-			err = runtime.DefaultUnstructuredConverter.FromUnstructured(r.Object, &pod)
-			if err != nil {
-				return err
-			}
-			pods[pod.GetName()] = struct{}{}
-			if podChan != nil {
-				podChan <- pod
-			}
-		}
+	resources, err := o.kubernetesClient.GetAllResourcesFromSelector(selector, namespace)
+	if err != nil {
+		return err
 	}
 
 	// get all pods in the namespace
-	podList, err := o.kubernetesClient.GetAllPodsInNamespace()
+	podList, err = o.kubernetesClient.GetAllPodsInNamespace()
```

Comment:
I am not sure I understand why you want to get a list of all the pods in the namespace when you already have a list of the pods that match a given selector. Shouldn't that be enough?

Reply:
> I am not sure I understand why you want to get a list of all the pods in the namespace when you already have a list of the pods that match a given selector. Shouldn't that be enough?

If a Pod is created for a Deployment in Deploy mode, it won't have the selector, and hence won't show up in the list of Pods matching the given selector.